apps/gradient_autoscheduler/GradientAutoscheduler.cpp

```diff
@@ -0,0 +1,1008 @@
+#include "Halide.h"
+
+#ifndef user_error
```

Comment:
We need a better answer here -- as a starting point, maybe copy apps/autoscheduler/Errors.h here so that the evilness is contained in a header. (But really as a separate PR we need to figure out how to consolidate.)

Reply:
Moved.